cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -194,48 +194,11 @@ func (o *ScaleUpOrchestrator) ScaleUp(
 
 	createNodeGroupResults := make([]nodegroups.CreateNodeGroupResult, 0)
 	if !bestOption.NodeGroup.Exist() {
-		oldId := bestOption.NodeGroup.Id()
-		createNodeGroupResult, aErr := o.processors.NodeGroupManager.CreateNodeGroup(o.autoscalingContext, bestOption.NodeGroup)
+		var scaleUpStatus *status.ScaleUpStatus
+		createNodeGroupResults, scaleUpStatus, aErr = o.CreateNodeGroup(bestOption, nodeInfos, schedulablePods, podEquivalenceGroups, daemonSets)
```

Comment:
nit: It would be a nice to have if we can separate some unit test which are core to the new refactored logic to directly call `CreateNodeGroup`, and leave the rest as is.

this will decouple the testing also internal bits of the new func.

Reply:
The unit tests that check this path are [TestScaleUpAutoprovisionedNodeGroup](https://github.com/kubernetes/autoscaler/blob/a3c8978e119f56a90a6f8406e76685c21cfecd51/cluster-autoscaler/core/scaleup/orchestrator/orchestrator_test.go#L1421) and  [TestScaleUpBalanceAutoprovisionedNodeGroups](https://github.com/kubernetes/autoscaler/blob/a3c8978e119f56a90a6f8406e76685c21cfecd51/cluster-autoscaler/core/scaleup/orchestrator/orchestrator_test.go#L1473). I would prefer to leave them as they are currently, because they test both the creation and the scale up.

Reply:
ack.